tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -336,13 +337,24 @@ def stack_new_level_core(self, X, y, models: List[AbstractModel], X_val=None, y_
                     return []
             else:
                 raise AssertionError(f'Stack level cannot be negative! level = {level}')
-            models = [
-                stacker_type(
-                    path=self.path, name=model.name + '_STACKER_l' + str(level), model_base=model, base_model_names=base_model_names,
+            invalid_model_names = set(self.get_model_names_all())
+            models_stack = []
+            for model in models:
+                # Ensure name is unique
+                model_stack_name = f'{model.name}_STACKER_l{level}'
+                num_increment = 2
+                while model_stack_name in invalid_model_names:  # Ensure name is unique
+                    model_stack_name = f'{model.name}_{num_increment}_STACKER_l{level}'
+                    num_increment += 1
+                invalid_model_names.add(model_stack_name)
+
+                model_stack = stacker_type(
```

Comment:
should we print some output (with unimportant logger level) to alert user of the new model name and that duplicate model exists?

Reply:
We can come back to this question at a later time if we find users being confused about names, but I think in general it should be fine, I don't expect users to care too much about the exact name of models, and if they are specifying them explicitly then its likely it won't be a duplicate.